### PR TITLE
[4.0] Cassiopeia Header/Banner Style

### PR DESCRIPTION
**Pull request by ciar4n**
Pull Request for Issue # .

### Summary of Changes
Header styling for the Cassiopeia frontend template

### Testing Instructions
Apply PR and check frontend, with and without banner module.

### Before
image
--

### After
image
--
image
--
image
--
image
--

### CODE REVIEW

**Comment by dgrammatiko** [comment]
@ciar4n nice, couple remarks here:
- Font loading shouldn't be done like that, ultra inefficient
- The template (IMHO) should have **only** positions as placeholders, eg for the menu we should have:
```html
<header>
<jdoc:include type="modules" name="menu" style="none" />
<header>
```

So all the specific implementation of the top menu (if someone wants to have a top menu 😏) should go to the menu override. That will make the base template extremely versatile and flexible to any design!

File: templates/cassiopeia/index.php [hunk 1]
```diff
@@ -84,39 +84,45 @@
...
-			<?php endif; ?>
-		</nav>
-	</header>
 
-	<?php if ($this->countModules('banner')) : ?>
-	<div class="container-banner full-width">
-		<jdoc:include type="modules" name="banner" style="xhtml" />
+				<?php if ($this->countModules('menu')) : ?>
+					<button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="<?php echo JText::_('TPL_CASSIOPEIA_TOGGLE'); ?>">
+						<span class="fa fa-bars"></span>
```
Comment (brianteeman) [inline templates/cassiopeia/index.php]: aria-hidden=true

File: templates/cassiopeia/scss/blocks/_header.scss [hunk 2]
```diff
@@ -122,8 +146,40 @@
       margin-left: 5px;
       font-size: $h1-font-size;
       color: lighten($cassiopeia-template-color, 10%);
+
```
Comment (brianteeman) [inline templates/cassiopeia/scss/blocks/_header.scss:156]: is this blank line needed?

File: templates/cassiopeia/index.php (hunk 1 above)
Comment (ciar4n) [reply templates/cassiopeia/index.php]: Done.. ty

File: templates/cassiopeia/scss/blocks/_header.scss (hunk 2 above)
Comment (ciar4n) [reply templates/cassiopeia/scss/blocks/_header.scss:156]: I don't think we have a definite codestyle yet for J4 SCSS. This is basically just following the format previously set by @C-Lodder where a blank line is added after each final declaration.

Comment (brianteeman) [reply templates/cassiopeia/scss/blocks/_header.scss:156]: Ah ok.

**Comment by C-Lodder** [comment]
Can you just confirm this image is from the image repo used by Joomla and not an image from Google? If so, then all good to merge

**Comment by ciar4n** [comment]
Image is from unsplash (open)... https://unsplash.com/license

**Comment by ciar4n** [comment]
> The template (IMHO) should have only positions as placeholders

@dgt41 Regarding this, currently both the menu and search module positions are behind the display toggle on smaller screens. Because of this I am not sure how feasible it is to move all markup to the override,

**Comment by dgrammatiko** [comment]
Let me explain what should be in the override:
```html
			<nav class="navbar navbar-expand-lg">
				<div class="navbar-brand">
					<a href="<?php echo $this->baseurl; ?>/">
						<?php echo $logo; ?>
					</a>
					<?php if ($this->params->get('siteDescription')) : ?>
						<div class="site-description"><?php echo htmlspecialchars($this->params->get('siteDescription')); ?></div>
					<?php endif; ?>
				</div>

				<?php if ($this->countModules('menu') || $this->countModules('search')) : ?>
					<button class="navbar-toggler navbar-toggler-right" type="button" aria-hidden="true" data-toggle="collapse" data-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="<?php echo JText::_('TPL_CASSIOPEIA_TOGGLE'); ?>">
						<span class="fa fa-bars"></span>
					</button>
					<div class="collapse navbar-collapse" id="navbar">
						<!-- !!! Here goes the menu output-->
						<?php if ($this->countModules('search')) : ?>
							<div class="form-inline">
								<!-- !!! Here goes the search module -->
							</div>
						<?php endif; ?>
					</div>
				<?php endif; ?>
				
			</nav>
			<?php if ($this->countModules('banner')) : ?>
			<div class="container-banner">
				<jdoc:include type="modules" name="banner" style="xhtml" />
			</div>
			<?php endif; ?>
			<div class="header-shadow"></div>
			<svg class="header-shape-bottom" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 736 15">
				<path d="M1040,301V285s-75,12-214,12-284-26-524,0v4Z" transform="translate(-302 -285)" fill="#fff"/>
			</svg>
```

**Pull request merged**